netcore/CMakeLists.txt

```diff
@@ -12,16 +12,15 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 project(netcore LANGUAGES NONE)
 
 option(INCLUDE_DOTNET_CORE "Build .NET Core client." ON)
 
 find_program(DOTNET dotnet)
 
-if(DOTNET AND INCLUDE_DOTNET_CORE)
-  add_custom_target(netcore ALL COMMAND ${DOTNET} build --configuration $<CONFIG> WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/NetCore DEPENDS apache-geode-c VERBATIM)
-  add_custom_target(netcore-test ALL COMMAND ${DOTNET} build --configuration $<CONFIG> WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/NetCore.Test DEPENDS netcore VERBATIM)
+add_subdirectory(NetCore)
```

Comment:
You should probably .NET Core check around the `add_subdirectory` calls. This prevents them from even being processed on when not enabled.

```
if(DOTNET AND INCLUDE_DOTNET_CORE)
endif()
```